src/main/java/duke/main/Duke.java

```diff
@@ -0,0 +1,160 @@
+package duke.main;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.TaskException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Parser parser;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = storage.load();
+        } catch (FileNotFoundException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Adds a new task depending on type and append in to tasklist.
+     *
+     * @param whole_str the whole input string.
+     * @param str the whole input string, split.
+     * @throws TaskException
+     */
+    private void addTask(String whole_str, String[] str) throws TaskException {
+        if (str[0].compareTo("deadline") == 0) {
```

Comment:
The complex expressions in the series of if statements make the code a little more difficult to read. Generic parameter names as well. Perhaps this could be refactored for better readability?